kedro/io/partitioned_data_set.py

```diff
@@ -439,7 +439,7 @@ def _parse_checkpoint_config(
         if self._credentials:
             default_config[CREDENTIALS_KEY] = deepcopy(self._credentials)
 
-        if CREDENTIALS_KEY in default_config.keys() & checkpoint_config.keys():
+        if CREDENTIALS_KEY in (default_config.keys() & checkpoint_config.keys()):
```

Comment:
I may be wrong, but I believe this condition will never be true. I did some tests and looks like the `&` operator returns an empty set. Please, see the screenshot below:

<img width="279" alt="Screen Shot 2021-12-03 at 15 16 41" src="https://user-images.githubusercontent.com/4855538/144652833-1f859d2f-cf02-4698-99cf-4d8075c6a4e8.png">

Reply:
@arnaldog12 Hey these are set-like views, so `&` only returns non empty set if the intersection is not empty.

![Screenshot 2021-12-03 at 18 21 14](https://user-images.githubusercontent.com/2032984/144653300-f7c727fb-d45f-438c-a6b2-deba78b583d9.png)

Reply:
I think there is a test that covers this case. In this PR I just want to make it compatible with the pylint release 40 minutes ago but I give up. I will file a ticket on pylint.